src/lib/Libutil/misc_utils.c

```diff
@@ -745,30 +746,28 @@ escape_delimiter(char *str, char *delim, char esc)
 		return strdup((char *)str);
 	}
 	delim_len = strlen(delim);
-	retstr = (char *)malloc(MAXPATHLEN);
-	if(retstr == NULL)
+	retstr = (char *) malloc(MAXBUFLEN);
+	if (retstr == NULL)
 		return NULL;
-	retstrlen = MAXPATHLEN;
+	retstrlen = MAXBUFLEN;
 
 	while (*str != '\0') {
-		if(*str == esc) {
+		if ((*str == esc && !(*(str + 1) == '\'' || *(str + 1) == '\"'))
```

Comment:
I like that you have optimized it for a single byte delimiter.
This if statement has now become very complex to understand though. You might want to add a comment to explain it.